zerver/lib/markdown/__init__.py

```diff
@@ -163,7 +163,7 @@ def verbose_compile(pattern: str) -> Pattern[str]:
 
 
 STREAM_LINK_REGEX = r"""
-                     (?<![^\s'"\(,:<])            # Start after whitespace or specified chars
+                     (?<![^\s'"\(\{\[\/<])            # Start after whitespace or specified chars
```

Comment:
Why did we remove `,` and `:`?

Reply:
That's because we expect there to be a space between `,` or `:` and the stream mention. This was discussed [here](https://chat.zulip.org/#narrow/stream/9-issues/topic/Mention.20after.20.2F)

Reply:
Cool, then it seems fine!

Reply:
@N-Shar-ma it looks like you didn't apply this change from the STREAM_TOPIC regex; is that an oversight?

Also, should we extract `BEFORE_MENTION_ALLOWED_REGEX = r"(?<![^\s'"\(\{\[\/<])"` in `mention.py` and have all of these other regular expressions be constructed using that? Unless I'm mistaken, we want the same thing in all 4 places, and we should express that in a way that indicates that.

Reply:
Oops, that was an oversight! Have extracted the variable for re-use as suggested!